src/bridge/bridge.c

```diff
@@ -746,10 +725,34 @@ main (int argc,
       return 0;
     }
 
-  if (!opt_interactive && isatty (1))
+  if (!opt_interactive)
     {
-      g_printerr ("cockpit-bridge: no option specified\n");
-      return 2;
+      if (isatty (1))
+        {
+          g_printerr ("cockpit-bridge: no option specified\n");
+          return 2;
+        }
+
+      /* Connect stderr to the local journal explicitly.  This makes
+       * sure that we log to the machine where this process is
+       * running, and that glib always does structured logging even if
+       * our stderr was redirected to a pipe to capture errors from
+       * any wrapper that has started us (like sudo).
+       */
+
+      int fd = sd_journal_stream_fd ("cockpit-bridge", LOG_WARNING, 0);
```

Comment:
In any case, please set this to something other than "cockpit-bridge": cockpit-bridge will end up with the glib-reopened channel and name itself correct "cockpit-bridge".  Everything else that ends up in here will be something which *isn't* cockpit-bridge (ie: random stderr output from the things we spawn).  I set this to `cockpit/ssh` before it make it clear that it wasn't a program name.  Maybe `cockpit/session` now or so?

Reply:
Ok, done!